playbooks/artifact_upload.yml

```diff
@@ -0,0 +1,54 @@
+---
+- name: Artifact uploader
+  hosts: localhost
+  connection: local
+  gather_facts: False
+  vars:
+    region: "DFW"
+    cloud_name: "public_cloud"
+    component_metadata_file_path: "{{ lookup('env', 'WORKSPACE') }}/component_metadata.yml"
+    legacy_artifact_log_type:
+      - "type": "log"
+        "source": "{{ lookup('env', 'RE_HOOK_ARTIFACT_DIR') | default(lookup('env', 'WORKSPACE') ~ '/artifacts', true) }}"
+  tasks:
+    - name: Check for component metadata file
+      stat:
+        path: "{{ component_metadata_file_path }}"
+        get_attributes: no
+        get_checksum: no
+        get_md5: no
+        get_mime: no
+      register: _component_metadata_file
+
+    - name: Import component metadata file if it exists
+      set_fact:
+        component_metadata: "{{ lookup('file', component_metadata_file_path) | from_yaml }}"
+      when:
+        - "_component_metadata_file.stat.exists | bool"
+
+    - name: Determine the list of artifacts to upload
+      set_fact:
+        component_artifacts: >-
+          {%- if (_component_metadata_file.stat.exists | bool) and (component_metadata.artifacts is defined) %}
+          {%-   if 'log' in component_metadata.artifacts | json_query('[*].type') %}
+          {%-     set _result = component_metadata.artifacts %}
+          {%-   else %}
+          {%-     set _result = component_metadata.artifacts + legacy_artifact_log_type %}
+          {%-   endif %}
+          {%- else %}
+          {%-   set _result = legacy_artifact_log_type %}
+          {%- endif %}
```

Comment:
I tried to keep the specifics (uncluding checking if the source exists) of each type in the task files for that type, rather than here. It may be slightly less efficient, but it makes it more obvious why an artifact did not get uploaded in the resulting console output because you're able to see that the type config was picked up (or not), but that it got skipped because the source did not exist. I figure that this is more useful for diagnosing issues than skipping it earlier where the cause of skipping would be less obvious.